forte/data/base_pack.py

```diff
@@ -430,20 +461,140 @@ def on_entry_creation(
             # Use the auto-inferred control component.
             c = self.__control_component
 
+        def entry_getter(cls: Entry, attr_name: str, field_type):
+            """A getter function for dataclass fields of entry object.
+            When the field contains ``tid``s, we will convert them to entry
+            object on the fly.
+            """
+            data_store_ref = (
+                cls.pack._data_store  # pylint: disable=protected-access
+            )
+            attr_val = data_store_ref.get_attribute(
+                tid=cls.tid, attr_name=attr_name
+            )
+            if field_type in (FList, FDict):
+                # Generate FList/FDict object on the fly
+                return field_type(parent_entry=cls, data=attr_val)
+            try:
+                # TODO: Find a better solution to determine if a field is Entry
+                # Will be addressed by https://github.com/asyml/forte/issues/835
+                # Convert tid to entry object on the fly
+                if isinstance(attr_val, int):
+                    # Single pack entry
+                    return cls.pack.get_entry(tid=attr_val)
+                elif (
+                    isinstance(attr_val, tuple)
+                    and len(attr_val) == 2
+                    and all(isinstance(element, int) for element in attr_val)
+                    and hasattr(cls.pack, "get_subentry")
+                ):
```

Comment:
There is quite a bit of check and condition here, let's document what happens here.